bioblend/galaxy/objects/wrappers.py

```diff
@@ -626,6 +627,62 @@ def delete(self):
         self.refresh()
 
 
+@six.add_metaclass(abc.ABCMeta)
+class DatasetCollection(Wrapper):
+    """
+    Abstract base class for Galaxy dataset collections.
+    """
+    BASE_ATTRS = Wrapper.BASE_ATTRS + (
+        'state', 'deleted'
+    )
+
+    @abc.abstractmethod
+    def __init__(self, dsc_dict, container, gi=None):
+        super(DatasetCollection, self).__init__(dsc_dict, gi=gi)
+        object.__setattr__(self, 'container', container)
+
+    def refresh(self):
+        """
+        Re-fetch the attributes pertaining to this object.
+
+        Returns: self
+        """
+        gi_client = getattr(self.gi.gi, self.container.API_MODULE)
+        dsc_dict = gi_client.show_dataset_collection(self.container.id, self.id)
+        self.__init__(dsc_dict, self.container, self.gi)
+        return self
+
+
+class HistoryDatasetCollectionAssociation(DatasetCollection):
+    """
+    Maps to a Galaxy ``HistoryDatasetCollectionAssociation``.
+    """
+    BASE_ATTRS = DatasetCollection.BASE_ATTRS + ('tags', 'visible', 'elements')
+    SRC = 'hdca'
+
+    def __init__(self, dsc_dict, container, gi=None):
+        super(HistoryDatasetCollectionAssociation, self).__init__(
+            dsc_dict, container, gi=gi)
+
+    @property
+    def gi_module(self):
+        return self.gi.histories
+
+    @property
+    def _stream_url(self):
```

Comment:
This method is not used and does not make sense for a dataset collection, please remove.